server/sandbox/create.go

```diff
@@ -0,0 +1,101 @@
+package sandbox
+
+import (
+	"context"
+)
+
+// Creator is the interface which reflects necessary steps in sandbox
+// creation. All methods have to be called in order to successfully creator a
+// pod sandbox.
+// TODO: add the necessary arguments to the methods
+type Creator interface {
+	// GenerateName prepares the name and unique identifier of the pod sandbox.
+	// It has to be ensured that the sandbox name does not already exist.
+	GenerateName()
+
+	// PullPauseImage ensures that the pause image of the container exists
+	// before container creation.
+	PullPauseImage()
+
+	// CreateContainer builds the container from the provided names,
+	// identifiers and pause image.
+	CreateContainer()
+
+	// SetupOCISpec prepares the basic, host specific OCI configuration,
+	// like setting the process args for the pause image execution or DNS
+	// configuration.
+	SetupOCISpec()
+
+	// AddMetadata adds the provided metadata to the sandbox.
+	AddMetadata()
+
+	// AddLabels adds the provided labels to the sandbox.
+	AddLabels()
+
+	// AddAnnotation adds the provided annotations to the sandbox.
+	AddAnnotations()
+
+	// AddCapabilities adds the provided capabilities to the sandbox.
+	AddCapabilities()
+
+	// SetupLogging ensures that the absolute logging path for the sandbox
+	// exists.
+	SetupLogging()
+
+	// SetupNamespaceOptions ensures that the provided namespace options are
+	// applied to the sandbox.
+	SetupNamespaceOptions()
+
+	// SetupSharedMemory applies the shared memory settings to the sandbox.
+	// This also mounts the shared memory into the container.
+	SetupSharedMemory()
+
+	// SetupPortMappings prepares the port mappings to the sandbox.
+	SetupPortMappings()
+
+	// SetupSysctls prepares the sysctls for the sandbox.
+	SetupSysctls()
+
+	// SetupCgroups ensures the correct parent cgroup for the sandbox.
+	SetupCgroups()
+
+	// SetupNetwork creates the network for the sandbox.
+	SetupNetwork()
```

Comment:
We should define well the relationship between setup network and setup namespace options, as network is one of the namespace options

Reply:
This is obsolete with the latest version of this PR.